app/Models/Airline.php

```diff
@@ -39,7 +39,10 @@ public function scopeFilterByCity(Builder $query, ?int $cityId): void
 
     public function scopeFilterByFlights(Builder $query, ?int $flights): void
     {
-        $query->when(isset($flights), function ($query) use ($flights) {
+        $hasFlightsCount = collect($query->getQuery()->columns)->contains('flights_count');
+
+        $query->when(isset($flights), function ($query) use ($flights, $hasFlightsCount) {
+            $query->when($hasFlightsCount, fn($query) => $query->withCount('flights'));
```

Comment:
shouldn't be `!$hasFlightsCount` ??

Reply:
Yes! Well spotted. I wasn't getting an error when running this code because another bug was present, and `$hasFlightsCount` wasn't adequately registering if the column had been included yet. This was because `$query->getQuery()->columns` isn't getting a list of the column names: . The best way I found to check if the column had been included was to get the raw SQL from the query builder instance and check if the column has been included.
So: `$hasFlightsCount = Str::contains($query->getQuery()->toSql(), 'flights_count');`